index.js

```diff
@@ -0,0 +1,365 @@
+//. # sanctuary-argv
+//.
+//. The JavaScript ecosystem has many libraries for parsing command-line
+//. arguments; sanctuary-argv differs from these in a few important ways.
+//.
+//. ## Philosophy
+//.
+//. sanctuary-argv is completely pure. It does not:
+//.
+//.   - reference global variables (such as `process.argv`);
+//.
+//.   - read from standard input;
+//.
+//.   - write to standard output or standard error;
+//.
+//.   - mutate its arguments; or
+//.
+//.   - throw exceptions.
+//.
+//. ## Terminology
+//.
+//.     cut -s -d : -f 1 -- file1 file2 file3 = command
+//.         -s -d : -f 1 -- file1 file2 file3 = arguments
+//.         -s                                = flags = flag names
+//.            -d : -f 1                      = options
+//.            -d   -f                        = option names
+//.               :    1                      = option values
+//.                      --                   = separator
+//.                         file1 file2 file3 = positional arguments
+//.
+//. ### Design decisions
+//.
+//.   - There are no implicit flags or options. If the command-line interface
+//.     is to support `--help`, for example, sanctuary-argv does not generate
+//.     the help text.
+//.
+//.   - Both long option names (e.g. `--extension`) and short option names
+//.     (e.g. `-x`) are supported.
+//.
+//.   - Short option names and long option names must be specified separately
+//.     (whereas [Commander.js][] supports `'-x, --extension <extension>'`).
+//.
+//.   - If a long option name has an associated value, the value must be
+//.     provided as a separate argument (e.g. `['--extension', '.js']`).
+//.     `--name=value`-style options are not supported.
+//.
+//.   - If a short option name has an associated value, the value must be
+//.     provided as a separate argument (e.g. `['-x', '.js']`).
+//.
+//.   - Short option names must be provided as separate arguments. One can
+//.     support `tar -z -x -v -f filename` but not `tar -zxvf filename`.
```

Comment:
Although I agree with the other design decisions, this one in particular seems like a loss. The way I see it, short form flags are a write-optimized system that exists only to improve the UX of the user of the command within the context of the terminal. Its purpose is to allow a command to be expressed in as few keystrokes as possible.

With the `space` key being all the way down there, and the `-` key all the way up there, potentially significant typing overhead is added to a system whose sole purpose it is to reduce typing overhead.

I'm curious about the reasoning behind this decision.

Reply:
Would you like to support every possible interpretation of `-xyz`? It would be short for one of the following canonical representations, based on the specification:

  - `-x yz`
  - `-x -y z`
  - `-x -y -z`

Squashing an option value into a group of option names has always felt strange to me. What do you think?

Reply:
Ah, I hadn't considered the `cut -f1` case, just the `tar -xvf` case. I mean, using it to save a single space seems less relevant than combining multiple flags, saving a space and a dash per flag. I'd still expect values to be given after a space. like in `set -eufo pipefail`, three flags and one option.

But I guess now we're in the realm of opinion.

Reply:
I'm willing to support `-xyz` if it is always short for `-x -y -z`, irrespective of the specification. :)